packages/execute-driver/lib/execute-child.js

```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash';
+import B from 'bluebird';
+import vm from 'vm';
+import { logger } from '@appium/support';
+import { attach } from 'webdriverio';
+
+const log = logger.getLogger('ExecuteDriver Child');
+
+// duplicate defining these keys here so we don't need to re-load a huge appium
+// dependency tree into memory just to run a wdio script
+export const W3C_ELEMENT_KEY = 'element-6066-11e4-a52e-4f735466cecf';
+export const MJSONWP_ELEMENT_KEY = 'ELEMENT';
+
+async function runScript (driverOpts, script, timeout) {
+  if (!_.isNumber(timeout)) {
+    throw new Error('Timeout parameter must be a number');
+  }
+
+  // set up fake logger
+  const logLevels = ['error', 'warn', 'log'];
+  const logs = {};
+  const consoleFns = {};
+  for (const level of logLevels) {
+    logs[level] = [];
+    consoleFns[level] = (...logMsgs) => logs[level].push(...logMsgs);
+  }
+
+  const driver = await attach(driverOpts);
+
+  const fullScript = buildScript(script);
+  // the timeout here will not matter really, but set it anyway to be on the
+  // safe side
+  const vmCtx = vm.runInNewContext(fullScript, {}, {timeout});
+
+  // run the driver script, giving user access to the driver object, a fake
+  // console logger, and a promise library
+  log.info('Running driver script in Node vm');
+  let result = await vmCtx(driver, consoleFns, B);
+
+  log.info('Ensuring driver script result is appropriate type for return');
+  result = coerceScriptResult(result);
+  return {result, logs};
+}
+
+/**
+ * Embed a user-generated script inside a method which takes only the
+ * predetermined objects we specify
+ *
+ * @param {string} script - the javascript to execute
+ *
+ * @return {string} - the full script to execute
+ */
+function buildScript (script) {
+  return `(async function execute (driver, console, Promise) {
+    ${script}
+  })`;
```

Comment:
that being said, we may want to pick up something like [vm2](https://npm.im) instead, which makes it less trivial

Reply:
also, the `timeout` may be important since a `script` like:

```
while (true) {}
```

would happily run forever

Reply:
(It's unclear to me if the `vm` module can actually halt execution of this)

Reply:
https://i.ytimg.com/vi/KEkrWRHCDQU/maxresdefault.jpg

Reply:
Maybe we could just `nullify` env for the forked process (e.g. set it to `{}`)? just because of being paranoidal

Reply:
`{}` has the same issue.  basically: you cannot provide any vanilla object (you have to do a lot of work to avoid it--which is what vm2 does) created in the parent context to the new vm context w/o leaking.

Reply:
does using vm2 address all these issues?

Reply:
...maybe.  it came close once upon a time, but I haven't used it in a few years.  it's certainly safer than `vm`, assuming it works with Node.js v12-16